libs/BLEKit/include/BLEServiceMonitoring.h

```diff
@@ -52,6 +61,8 @@ class BLEServiceMonitoring : public interface::BLEService
 
 	void onSoftReboot(const std::function<void()> &callback) { _on_soft_reboot = callback; }
 
+	void registerGoToSleepRequested(const std::function<void()> &callback) { _on_go_to_sleep_callback = callback; }
```

Comment:
in English "register go to sleep requested" doesn't mean much.

I would say "register 'on go to sleep' callback"

question is: is the callback called:

- every time the app requested the robot to go to sleep?
- every time the robot actually goes to sleep?

is it possible for the request to no succeed? meaning that the robot will not go to sleep?

in this case, is the callback called?

to draw a parallel with iOS, a lot of Apple API have a notion of "tense":

- before action --> "view WILL appear" - `viewWillAppear`
- after action --> "view DID load" - `viewDidLoad`

it's important to know, from the API and the naming, if and when the registered callback will be called.

Taking this into account we could have:

```suggestion
	void registerWillGoToSleepCallback(const std::function<void()> &callback) { _on_go_to_sleep_callback = callback; }
```

```suggestion
	void registerDidGoToSleepCallback(const std::function<void()> &callback) { _on_go_to_sleep_callback = callback; }
```